system-contracts/contracts/L1Messenger.sol

```diff
@@ -313,25 +315,21 @@ contract L1Messenger is IL1Messenger, SystemContractBase {
             revert ReconstructionMismatch(PubdataField.InputLogsRootHash, localLogsRootHash, inputChainedLogsRootHash);
         }
 
-        bytes32 l2DAValidatorOutputhash = bytes32(0);
-        if (_l2DAValidator != address(0)) {
-            bytes memory returnData = EfficientCall.call({
-                _gas: gasleft(),
-                _address: _l2DAValidator,
-                _value: 0,
-                _data: _operatorInput,
-                _isSystem: false
-            });
-
-            l2DAValidatorOutputhash = abi.decode(returnData, (bytes32));
-        }
+        // Validate pubdata and make commitment. Logs are not checked since we already checked them above.
+        // Does nothing if commitment scheme is EMPTY_NO_DA.
+        bytes32 l2DAValidatorOutputhash = L2DAValidator.makeDACommitment(
+            _l2DACommitmentScheme,
+            inputChainedMsgsHash,
+            inputChainedBytecodesHash,
+            _operatorInput[4 + 32 * 4 + 64:] // Operator data offset (as bytes slice)
```

Comment:
Can we remove selector and maybe logs hash from `operatorInput`?

Reply:
We can, but beauty of this version is that we don't need to change an encoding from operator side. But if it will make your changes easier later, I can do it

Reply:
Please add comments on why these fields are still there in the pubdata